tests/Http/AcceptHeaderTest.php

```diff
@@ -0,0 +1,163 @@
+
+        $item = $item->withParams(['n' => 'new']);
+
+        $this->assertEquals(['n' => 'new'], $item->getParams());
+
+        $wrongParamsItem = AcceptHeaderItem::fromString('text/html; q');
+
+        $this->assertEquals([], $wrongParamsItem->getParams());
+        $this->assertEquals(1.0, $wrongParamsItem->getQuality());
+
+        AcceptHeaderItem::fromString('');
+    }
+
+    public function testHeaderConstructing()
+    {
+        $acceptCharset = new AcceptHeader(['*', 'UTF-8']);
+
+        $this->assertFalse($acceptCharset->has('unicode'));
+        $this->assertNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('*', $sorted[1]);
+        $this->assertEquals('UTF-8, *', $acceptCharset);
+
+        $acceptCharset = $acceptCharset->add(AcceptHeaderItem::fromString('unicode'));
+
+        $this->assertTrue($acceptCharset->has('unicode'));
+        $this->assertNotNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('unicode', $sorted[1]);
+        $this->assertEquals('*', $sorted[2]);
+        $this->assertEquals('UTF-8, unicode, *', $acceptCharset);
+
+        $acceptHeader = AcceptHeader::fromString('text/html; q=0.8, */*; q=0.7');
+        $acceptHeader = $acceptHeader->add('application/json');
+
+        $sorted = $acceptHeader->sorted();
+        $this->assertEquals('application/json', $sorted[0]);
+        $this->assertEquals('text/html; q=0.8', $sorted[1]);
+        $this->assertEquals('*/*; q=0.7', $sorted[2]);
+    }
+
+    public function testHeader1()
+    {
+        $acceptHeader = AcceptHeader::fromString('audio/*; q=0.2, audio/basic')->sorted();
+        $this->assertEquals('audio/basic', (string) $acceptHeader[0]);
+        $this->assertEquals('audio/*; q=0.2', (string) $acceptHeader[1]);
+    }
+
+    public function testHeader2()
+    {
+        $acceptHeader = AcceptHeader::fromString(
+            'text/*;q=0.3, text/html;q=0.7, text/html;level=1, text/html;level=2;q=0.4, */*;q=0.5'
+        )->sorted();
+
+        $this->assertEquals('text/html; level=1', $acceptHeader[0]);
+        $this->assertEquals('text/html; q=0.7', $acceptHeader[1]);
+        $this->assertEquals('*/*; q=0.5', $acceptHeader[2]);
+        $this->assertEquals('text/html; q=0.4; level=2', $acceptHeader[3]);
+        $this->assertEquals('text/*; q=0.3', $acceptHeader[4]);
+    }
+
+    public function testHeader3()
+    {
+        $acceptHeader = AcceptHeader::fromString('text/*, text/html, text/html;level=1, */*')->sorted();
+
+        $this->assertEquals('text/html; level=1', $acceptHeader[0]);
+        $this->assertEquals('text/html', $acceptHeader[1]);
+        $this->assertEquals('text/*', $acceptHeader[2]);
+        $this->assertEquals('*/*', $acceptHeader[3]);
+    }
+
```

Comment:
What will happen if the accept header will have empty items? The next tese case will fail
```suggestion
    public function testHeaderSanitize(): void
    {
        $acceptHeader = AcceptHeader::fromString('text/*, text/html, ,;,text/html;level=1, */*')->sorted();

        $this->assertCount(4, $acceptHeader);
    }
```